File: AddTwoNumbers.java
import java.util.ArrayList;

import java.util.List;
class AddTwoNumbers{
public static void main(final String[] args) {
    public final ListNode addTwoNumbers(final ListNode l1, final ListNode l2) {
        final ArrayList<Integer> num1 = new ArrayList<>();
        final ArrayList<Integer> num2 = new ArrayList<>();
        while(l1.next != null){
            num1.add(0, l1.val);
            l1 = l1.next;
        }
        while(l2.next != null){
            num2.add(0, l2.val);
            l2 = l2.next;
        }
        int n1 = Integer.parseInt(num1.toString());
        int n2 = Integer.parseInt(num2.toString());
        String sum = Integer.toString(n1 + n2);
        char[] array = sum.toCharArray();
        ListNode start = new ListNode((int)array[array.length -1]);
        ListNode node = start;
        for(int i = array.length -2 ; i  >= 0; i --){
            ListNode tmp = new ListNode((int)array[i]);
            node.next = tmp;
            node = node.next;
        }
    }
}
public class ListNode {
         int val;
         ListNode next;
         ListNode() {}
         ListNode(final int val) { this.val = val; }
         ListNode(final int val, final ListNode next) { this.val = val; this.next = next; }
     }
}

File: Add Two Numbers.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        tmp = l1
        num1 = []
        while tmp.next != None:
            num1.insert(0, str(tmp.val))
            tmp = tmp.next
        num1.insert(0, str(tmp.val))
        num1 = ''.join(num1)
        
        tmp = l2
        num2 = []
        while tmp.next != None:
            num2.insert(0, str(tmp.val))
            tmp = tmp.next
        num2.insert(0, str(tmp.val))
        num2 = ''.join(num2)
        total = (int(num1) + int(num2))
        result = [int(x) for x in str(total)]
        print(result)
        
        tmp = None
        for i in range(len(result)):
            now = ListNode(result[i], None)
            if tmp != None:
                now.next = tmp
                tmp = now
            else:
                tmp = now
        return (tmp)